Content.Shared/Doors/Systems/SharedDoorSystem.cs

```diff
@@ -536,6 +541,24 @@ public override void Update(float frameTime)
 
             if (door.NextStateChange.Value < time)
                 NextState(door, time);
+
+            if (door.State == DoorState.Closed &&
+                TryComp<PhysicsComponent>(door.Owner, out var doorBody))
+            {
+                // If something bumped into us during closing then start to re-open, otherwise, remove it from active.
+                if (doorBody.ContactCount == 0)
+                {
+                    _activeDoors.Remove(door);
+                }
+                else if (door.BumpOpen)
+                {
+                    foreach (var other in _physicsSystem.GetCollidingEntities(doorBody))
+                    {
```

Comment:
The body already has a linked list of contacts so this should be fast to iterate.